File: Problems/sorting/1365-how-many-numbers-are-smaller-than-the-current-number/main.c
#include <stdio.h>
#include <stdlib.h>

int* smallerNumbersThanCurrent(int* nums, int numsSize, int* returnSize) {
    int *result = malloc(sizeof(int) * *returnSize);

    for (int i=0; i<numsSize; i++) {
        for (int j=0; j<numsSize; j++) {
            if (i != j && nums[j] < nums[i]) {
                result[i]++;
            }
        }
    }

    return result;
}

void printArrayElements(int *arr, int arrSize) {
    for (int i=0; i<arrSize; i++) {
        printf("%d, ", arr[i]);
    }
    printf("\n");
}

int main() {
    int n = 7;
    int arr[] = {2, 4, 1, 6, 3, 8, 10};
    printArrayElements(arr, n);
    printArrayElements(smallerNumbersThanCurrent(arr, n, &n), n);

    return 0;
}

File: Problems/sorting/1365-how-many-numbers-are-smaller-than-the-current-number/main.py
def smallerNumbersThanCurrent(nums: list[int]) -> list[int]:
    numsSize = len(nums)
    result = [0 for i in range(numsSize)]
    for i in range(numsSize):
        for j in range(numsSize):
            if i != j and nums[j] < nums[i]:
                result[i] += 1

    return result


print(smallerNumbersThanCurrent([2, 4, 1, 6, 3, 8, 10]))
